Dockerfile

```diff
@@ -0,0 +1,47 @@
+# Build Hapistrano
+FROM alpine:3.7 as build-env
+
+MAINTAINER Javier Casas <jcasas@stackbuilders.com>
+
+RUN echo '@testing http://dl-cdn.alpinelinux.org/alpine/edge/testing' >> /etc/apk/repositories
+RUN apk update \
+ && apk add \
+        alpine-sdk \
+        bash \
+        ca-certificates \
+        cabal@testing \
+        ghc-dev@testing \
+        ghc@testing \
+        git \
+        gmp-dev \
+        gnupg \
+        libffi-dev \
+        linux-headers \
+        upx@testing \
+        zlib-dev
+
+WORKDIR /hapistrano
+
+COPY hapistrano.cabal .
```

Comment:
Better to `git clone` the repo, as we need `.git` when compiling, to fix `hap --version` as currently it doesn't show the commit and branch; there would be also less steps.

Reply:
That would checkout the last version, and would make it difficult to construct older versions. I'll see if I can copy the .git inside in order to be able to fix that.